src/database_bridge.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"reflect"
+	"strings"
+
+	_ "github.com/lib/pq"
+)
+
+var postgresCredentials string
+
+func init() {
+	b, err := ioutil.ReadFile("config/postgres_credentials.txt")
+	panicIfError(err)
+	postgresCredentials = string(b)
+
+	cutIndex := len(postgresCredentials) - 1
+	if postgresCredentials[cutIndex] == '\n' {
+		postgresCredentials = postgresCredentials[:cutIndex]
+	}
+}
+
+func createConnection() (*sql.DB, error) {
+	return sql.Open("postgres", postgresCredentials)
+}
+
+// Sends `query` to the database and writes queried results to `result` (which should be <b>a pointer</b> to either a
+// slice (in this case the queried data is appended to its end) or a struct (then only the first selected value is
+// written to it, others are suppressed)). If `result` is a pointer to a struct, but no results are returned by the
+// query, the appropriate error is returned, `result` is not updated
+func dbQuery(result interface{}, query string, args ...interface{}) error {
```

Comment:
As i know, sql.Open(...) creates a connection pool and you shouldn't re-create it every time. Check it please.

Reply:
You are right, I'll fix it